### PR TITLE
Add AUR package

### PR DESCRIPTION
**Pull request by crstian19**
**What**: Added AUR Package

**Why**: It can be installed from the AUR directly.

**How**: Creating the PKGBUILD

**Checklist**:

- [x] Documentation added
- [ ] Tests added
- [ ] No linting errors / broken tests
- [x] Merge ready

### CODE REVIEW

**Comment by coloradocolby** [comment]
@crstian19 this is awesome! I'll test it on my arch setup tonight. I'm not familiar with what the process of creating this looks like -- did you have to do anything outside of this project to set this up?

File: README.md [hunk 1]
```diff
@@ -25,6 +25,16 @@ $ cargo install thokr
 $ docker run -it coloradocolby/thokr
 ```
 
+## Arch Linux
+
+On Arch Linux, you can install it from AUR:
+
+``` bash
```
Comment (coloradocolby) [inline README.md:32]:
```suggestion
``` sh
```

File: README.md [hunk 2]
```diff
@@ -25,6 +25,16 @@ $ cargo install thokr
...
+paru -S thokr-git
```
Comment (coloradocolby) [inline README.md:33]: Ideally I'd like if the recommended way to download somehow didn't assume the user is using paru, yay, etc. Let me think about how to do this 🤔 I'll see if I can find examples in other repos I like!

File: README.md [hunk 3]
```diff
@@ -25,6 +25,16 @@ $ cargo install thokr
 $ docker run -it coloradocolby/thokr
 ```
 
+## Arch Linux
+
+On Arch Linux, you can install it from AUR:
```
Comment (coloradocolby) [inline README.md:30]: nit: I kinda like keeping READMEs minimal and to the point. Cargo and Docker don't have descriptions -- if someone is running arch I'm thinking this is probably not a necessary comment.

**Review by coloradocolby — COMMENTED** [review]
@crstian19 this is awesome! I'll test it on my arch setup tonight. I'm not familiar with what the process of creating this looks like -- did you have to do anything outside of this project to set this up?

File: README.md (hunk 3 above)
Comment (crstian19) [reply README.md:30]: It's ok, I'm agree

**Comment by crstian19** [comment]
> @crstian19 this is awesome! I'll test it on my arch setup tonight. I'm not familiar with what the process of creating this looks like -- did you have to do anything outside of this project to set this up?

I had to create the PKGBUILD in the AUR, basically it downloads the repository and uses cargo to do the installation.

This PKGBUILD is "-git" so it will use the last commit to update, the moment there is a release, I can create another package for stable versions.

File: README.md (hunk 2 above)
Comment (crstian19) [reply README.md:33]: Ok no problem, I have put paru because I think it is the one most people use right now (including me).

**Comment by coloradocolby** [comment]
> This PKGBUILD is "-git" so it will use the last commit to update, the moment there is a release, I can create another package for stable versions.

Can you explain what you mean by release? I have releases in gh already, though they're not exactly binaries. Would you need that? Also would you have to manually do this for every release, or is this a one time setup?

**Comment by crstian19** [comment]
> > This PKGBUILD is "-git" so it will use the last commit to update, the moment there is a release, I can create another package for stable versions.
> 
> Can you explain what you mean by release? I have releases in gh already, though they're not exactly binaries. Would you need that? Also would you have to manually do this for every release, or is this a one time setup?
> 
> Edit: they are pre-releases though -- maybe that's what you mean

Right now as you say you have pre-releases, when there is a stable release for example v1.0.0, simply take that release, it is not necessary to make a binary, everything is automatic.

**Comment by coloradocolby** [comment]
> Right now as you say you have pre-releases, when there is a stable release for example v1.0.0, simply take that release, it is not necessary to make a binary, everything is automatic.

Until there is a stable release, will the PKGBUILD need to be updated in the AUR each time there is a minor/patch update? Once there is a stable release, will it auto track my updates? What I'm getting at is what part of this process is automated and what is manual moving forward.

**Comment by crstian19** [comment]
> > Right now as you say you have pre-releases, when there is a stable release for example v1.0.0, simply take that release, it is not necessary to make a binary, everything is automatic.
> 
> Until there is a stable release, will the PKGBUILD need to be updated in the AUR each time there is a minor/patch update? Once there is a stable release, will it auto track my updates? What I'm getting at is what part of this process is automated and what is manual moving forward.

Each person who decides to install thokr-git knows that it will be the last commit and may not be stable.

The whole process is automatic, you will not have to do anything, for stable versions it will take the releases and for non-stable versions the last commit.

**Comment by coloradocolby** [comment]
Perfect. Thank you for the context on that. I think I like the way [gotop](https://github.com/cjbassi/gotop#arch-linux) handles this in their README, by simply stating:

## Arch Linux
Install `gotop`, `gotop-bin`, or `gotop-git` from the AUR.


If I understand you correctly, `thokr-git` will always pull the latest commit, `thokr` will pull the latest stable release (once we have one -- does it also look at github for this, or pull the crate directly)? If you could send me any documentation on what you're doing I'd be very curious to learn more!

**Comment by coloradocolby** [comment]
Also I'm happy to be the maintainer of it in the AUR if you'd like -- I'm not sure if you can even transfer it

**Comment by crstian19** [comment]
> Perfect. Thank you for the context on that. I think I like the way [gotop](https://github.com/cjbassi/gotop#arch-linux) handles this in their README, by simply stating:
> ## Arch Linux
> 
> Install `gotop`, `gotop-bin`, or `gotop-git` from the AUR.
> 
> If I understand you correctly, `thokr-git` will always pull the latest commit, `thokr` will pull the latest stable release (once we have one -- does it also look at github for this, or pull the crate directly)? If you could send me any documentation on what you're doing I'd be very curious to learn more!

Yes, it's just like you say. You can read all about uploading packages to the AUR here: https://wiki.archlinux.org/title/AUR_submission_guidelines

**Comment by crstian19** [comment]
> Also I'm happy to be the maintainer of it in the AUR if you'd like -- I'm not sure if you can even transfer it

For the time if it's ok with you I can give you co-mantainer of the git package until you have a little more experience on how to handle the AUR.

You will be able to modify everything you need and if there is any problem I could fix it.

I just need your AUR user.

**Comment by coloradocolby** [comment]
Sounds good! I'll send you a message at that point -- thanks for the add! I'll cut a release in a bit, gonna make some small changes myself :)

**Pull request merged**